csharp/src/Google.Protobuf/CodedInputReader.cs

```diff
@@ -0,0 +1,879 @@
+                    ThrowEndOfInputUnless(reader.TryRead(out value));
+                    tmp = value;
+                    if (tmp < 128)
+                    {
+                        result |= tmp << 21;
+                    }
+                    else
+                    {
+                        result |= (tmp & 0x7f) << 21;
+                        ThrowEndOfInputUnless(reader.TryRead(out value));
+                        tmp = value;
+                        result |= tmp << 28;
+                        if (tmp >= 128)
+                        {
+                            // Discard upper 32 bits.
+                            // Note that this has to use ReadRawByte() as we only ensure we've
+                            // got at least 5 bytes at the start of the method. This lets us
+                            // use the fast path in more cases, and we rarely hit this section of code.
+                            for (int i = 0; i < 5; i++)
+                            {
+                                ThrowEndOfInputUnless(reader.TryRead(out value));
+                                tmp = value;
+                                if (tmp < 128)
+                                {
+                                    return (uint)result;
+                                }
+                            }
+                            throw InvalidProtocolBufferException.MalformedVarint();
+                        }
+                    }
+                }
+            }
+            return (uint)result;
+        }
+
+        /// <summary>
+        /// Reads a raw varint from the stream.
+        /// </summary>
+        internal ulong ReadRawVarint64()
+        {
+            int shift = 0;
+            ulong result = 0;
+            while (shift < 64)
+            {
+                ThrowEndOfInputUnless(reader.TryRead(out byte b));
+                result |= (ulong)(b & 0x7F) << shift;
+                if ((b & 0x80) == 0)
+                {
+                    return result;
+                }
+                shift += 7;
+            }
+            throw InvalidProtocolBufferException.MalformedVarint();
+        }
+
+        /// <summary>
+        /// Reads a 32-bit little-endian integer from the stream.
+        /// </summary>
+        internal uint ReadRawLittleEndian32()
+        {
+            const int length = 4;
+
+            if (reader.CurrentSpan.Length - reader.CurrentSpanIndex >= length)
+            {
+                // Fast path. All data is in the current span.
+                ReadOnlySpan<byte> data = reader.CurrentSpan.Slice(reader.CurrentSpanIndex, length);
+                reader.Advance(length);
+
+                return BinaryPrimitives.ReadUInt32LittleEndian(data);
+            }
+            else
+            {
+                return ReadRawLittleEndian32Slow();
+            }
+        }
+
+        private unsafe uint ReadRawLittleEndian32Slow()
+        {
+            const int length = 4;
```

Comment:
Nit: Why local const int?

Reply:
Its used multiple times in the method, and using a variable is more readable that a magic number